velox/expression/Expr.cpp

```diff
@@ -334,38 +334,34 @@ class ExprExceptionContext {
       return;
     }
 
-    auto dataPath = generateFilePath(basePath, "vector");
-    auto sqlPath = generateFilePath(basePath, "sql");
-    if (!dataPath.has_value()) {
-      dataPath_ = "Failed to create file for saving input vector.";
-      return;
-    }
-    if (!sqlPath.has_value()) {
-      sqlPath_ = "Failed to create file for saving expression SQL.";
-      return;
-    }
-
     // Persist vector to disk
     try {
-      std::ofstream outputFile(dataPath.value(), std::ofstream::binary);
-      saveVector(*vector_, outputFile);
-      outputFile.close();
-      dataPath_ = dataPath.value();
-    } catch (...) {
-      dataPath_ =
-          fmt::format("Failed to save input vector to {}.", dataPath.value());
+      auto dataPathOpt = generateFilePath(basePath, kVectorPathPrefix.data());
+      if (!dataPathOpt.has_value()) {
+        dataPath_ = "Failed to create file for saving input vector.";
+        return;
+      }
+      dataPath_ = dataPathOpt.value();
+      saveVectorToFile(vector_, dataPath_.c_str());
+    } catch (std::exception& e) {
+      dataPath_ = e.what();
+      return;
     }
 
     // Persist sql to disk
+    auto sql = expr_->toSql();
     try {
-      auto sql = expr_->toSql();
-      std::ofstream outputFile(sqlPath.value(), std::ofstream::binary);
-      outputFile.write(sql.data(), sql.size());
-      outputFile.close();
-      sqlPath_ = sqlPath.value();
-    } catch (...) {
-      sqlPath_ =
-          fmt::format("Failed to save expression SQL to {}.", sqlPath.value());
+      auto sqlPathOpt = generateFilePath(basePath, kStringPathPrefix.data());
+      if (!sqlPathOpt.has_value()) {
+        // dataPath_ is used to deliver error message.
```

Comment:
Why is this change? Let's use sqlPath_

Reply:
I somehow remember the original implementation uses dataPath for error message. But since changes have been made and it does not make sense to have datapath to store the error message anymore. This change is wrong here in the commit. I'll revert it back